File: frontend/src/types/ai.ts
export type GearKind = "camera" | "lens";

export type GearLinkConfidence = "low" | "medium" | "high";

export type GearLinkSuggestion = {
    title: string;
    url: string;
    reason: string;
    confidence: GearLinkConfidence;
}

export type GearLinkRequest = {
    kind: GearKind;
    name: string;
}

export type GearLinkResponse = {
    kind: GearKind;
    name: string;
    summary: string;
    suggestions: GearLinkSuggestion[];
}

File: backend/internal/dto/ai.go
package dto

type GearLinkRequest struct {
	Kind string `json:"kind"`
	Name string `json:"name"`
}

type GearLinkSuggestion struct {
	Title      string `json:"title"`
	URL        string `json:"url"`
	Reason     string `json:"reason"`
	Confidence string `json:"confidence"`
}

type GearLinkResponse struct {
	Kind        string               `json:"kind"`
	Name        string               `json:"name"`
	Summary     string               `json:"summary"`
	Suggestions []GearLinkSuggestion `json:"suggestions"`
}
